pkg/operator/deploy/deploy.go

```diff
@@ -175,10 +175,9 @@ func (o *operator) resources() ([]kruntime.Object, error) {
 		},
 	}
 
-	// TODO (BV): reenable gateway once we fix bugs
-	// if o.oc.Properties.NetworkProfile.GatewayPrivateEndpointIP != "" {
-	// 	cluster.Spec.GatewayDomains = append(o.env.GatewayDomains(), o.oc.Properties.ImageRegistryStorageAccountName+".blob."+o.env.Environment().StorageEndpointSuffix)
-	// }
+	if o.oc.Properties.FeatureProfile.GatewayEnabled && o.oc.Properties.NetworkProfile.GatewayPrivateEndpointIP != "" {
+		cluster.Spec.GatewayDomains = append(o.env.GatewayDomains(), o.oc.Properties.ImageRegistryStorageAccountName+".blob."+o.env.Environment().StorageEndpointSuffix)
```

Comment:
Can we also clear the GatewayDomains if the gateway is not enabled?

Reply:
just to be clear on this - the request is to set `cluster.Spec.GatewayDomains` to an empty array? I'm assuming this would cover the case where we admin-disable the feature, so we want to also update dnsmask on each node. Now I'm also curious about `append` here in general, should it be a hard assignment instead? There are no other references to this that I can find in the codebase, but I might be missing something.

Reply:
I added an `else` not picked up by GH. Thumbs up to resolve?